mongodbatlasv2/api_access_tracking.go

```diff
@@ -115,6 +115,17 @@ func (r ListAccessLogsByClusterNameApiRequest) Execute() (*MongoDBAccessLogsList
 	return r.ApiService.ListAccessLogsByClusterNameExecute(r)
 }
 
+func (r ListAccessLogsByClusterNameApiRequest) ExecuteWithParams(params *ListAccessLogsByClusterNameApiParams) (*MongoDBAccessLogsList, *http.Response, error) {
+	r.groupId = params.GroupId 
+	r.clusterName = params.ClusterName 
+	r.authResult = params.AuthResult 
```

Comment:
similar to https://github.com/mongodb/go-client-mongodb-atlas/pull/464/files#r1174923339 this is a perfect case to test this PR
`authResult` when `true` means successful access, `false` means unsuccessful access and not set means both, can this method needs to cover the three cases

Reply:
In nonatlas specific cases `default` means that *if field is not present* backend will assume that value
This means that default should never be passed from the client (and even if they are they are redundant).

I have made a recent fix to pass the default values from the client for query params (which would never have landed upstream due to the above problem). Do we want to incorporate that here as well

Since we setting values checking for non nil values makes no sense (as it will end up with the same result anyway)

Reply:
Let's clarify it because we might have misunderstood what default behavior we want from client. 
For me, default is clearly informational for the end user and should not be present in the client.
The reality is that our defaults might not be applied by some of the atlas endpoints right? so that is why we need to pas them?

Reply:
Done by enabling *bool values:

- true,false
- Nil (apply default when sending request or ignore when  omitEmpty is present

Reply:
>Nil (apply default when sending request or ignore when omitEmpty is present

Do you have an example of omitempty? I can't see it on the template and can't tell if it's really in use

>Done by enabling *bool

What about string("") and numbers(0)

Reply:
> Do you have an example of omitempty? I can't see it on the template and can't tell if it's really in use

My bad.
Omit empty would be only for request body objects. Params objects are intermediate objects to hold arguments, they aren't used in serialization.

> What about string("") and numbers(0)

String and numbers are pointers (as internal structs also use pointers), thus solving the problem of uncertainty if the value is provided. There would be also outstanding work on the atlas cli to have flags that do provide empty values (instead of 0 or math.MaxInt32 it can be nil)